terraform/modules/cloudwatch_dashboard/main.tf

```diff
@@ -8,8 +8,8 @@ resource "aws_cloudwatch_dashboard" "dashboard" {
         properties : {
           metrics : [
             for metric in widget.metrics : flatten([
-              metric.name == null ? [] : var.metric_namespace,
-              metric.name == null ? [] : metric.name,
+              metric.name == null ? [] : [var.metric_namespace],
+              metric.name == null ? [] : [metric.name],
```

Comment:
This is feeling a bit messy now and maybe slightly over utilising flatten, not sure if there's a nicer way of doing something like this?